app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-body.js

```diff
@@ -10,55 +11,53 @@ export default Component.extend({
     return false;
   }),
 
-  rootEventType: "click",
-
-  init() {
-    this._super(...arguments);
-
-    this.handleRootMouseDownHandler = bind(this, this.handleRootMouseDown);
-  },
-
   didInsertElement() {
     this._super(...arguments);
 
     this.element.style.position = "relative";
-
-    document.addEventListener(
-      this.rootEventType,
-      this.handleRootMouseDownHandler,
-      true
-    );
+    document.addEventListener("click", this.handleClick, true);
+    this.selectKit
+      .mainElement()
+      .addEventListener("keydown", this._handleKeydown, true);
   },
 
   willDestroyElement() {
     this._super(...arguments);
-
-    document.removeEventListener(
-      this.rootEventType,
-      this.handleRootMouseDownHandler,
-      true
-    );
+    document.removeEventListener("click", this.handleClick, true);
+    this.selectKit
+      .mainElement()
+      ?.removeEventListener("keydown", this._handleKeydown, true);
   },
 
-  handleRootMouseDown(event) {
-    if (!this.selectKit.isExpanded) {
+  @bind
+  handleClick(event) {
+    if (!this.selectKit.isExpanded || !this.selectKit.mainElement()) {
       return;
     }
 
-    const headerElement = document.querySelector(
-      `#${this.selectKit.uniqueID}-header`
-    );
-
-    if (headerElement && headerElement.contains(event.target)) {
+    if (this.selectKit.mainElement().contains(event.target)) {
       return;
     }
 
-    if (this.element.contains(event.target)) {
+    this.selectKit.close(event);
+  },
+
+  @bind
+  _handleKeydown(event) {
+    if (!this.selectKit.isExpanded || event.key !== "Tab") {
       return;
     }
 
-    if (this.selectKit.mainElement()) {
+    later(() => {
+      if (
+        this.isDestroying ||
+        this.isDestroyed ||
+        this.selectKit.mainElement()?.contains(document.activeElement)
+      ) {
+        return;
+      }
+
       this.selectKit.close(event);
-    }
+    }, 50);
```

Comment:
This seems like it has the potential to be flaky... what's the significance of 50ms here? (I suspect this delay is the reason that `selectKit.mainElement()` returned null and caused errors)

Reply:
This is not ideal, agreed. The reason for this delay is that when during `keyDown` the activeElement hasn't changed yet, so we need to wait a bit and then check. I'm looking at alternatives... 

That said, this is not the source of the errors. There was no keydown event happening at all during these errors, I can repro the issue locally even when I comment out the full `later()` block. The issue is that the event is assigned to the parent of `select-kit-body` and calling it via `this.selectKit.mainElement()` seems to be slow, the parent's already removed by then.

Reply:
> during keyDown the activeElement hasn't changed yet

I see :+1:. I wonder if we could use `next()` instead? If it works, that feels a little cleaner than an arbitrary delay.

Reply:
Yes, just switched to `next()`, it works well!

Reply:
I'd like to try merging this again on my tomorrow morning @davidtaylorhq, if there are no further objections. Thanks!